src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -180,6 +197,44 @@ private static void WriteFunctionsMetricEvent(string executionId, ulong executio
             {
                 MetricEventSource.Log.RaiseFunctionsMetricEvent(executionId, executionTimeSpan, executionCount, executionStage);
             }
+
+            private void RaiseMetricsPerFunctionEvent()
+            {
+                List<FunctionMetrics> metricsEventsList = GetMetricsQueueSnapshot();
+
+                var aggregatedEventsPerFunction = from item in metricsEventsList
+                                                  group item by new { item.FunctionName } into FunctionGroups
+                                                  select new
+                                                  {
+                                                      FunctionName = FunctionGroups.Key.FunctionName,
+                                                      StartedCount = Convert.ToUInt64(FunctionGroups.Count(x => x.ExecutionStage == ExecutionStage.Started)),
+                                                      FailedCount = Convert.ToUInt64(FunctionGroups.Count(x => x.ExecutionStage == ExecutionStage.Failed)),
+                                                      SucceededCount = Convert.ToUInt64(FunctionGroups.Count(x => x.ExecutionStage == ExecutionStage.Succeeded)),
+                                                      TotalExectionTimeInMs = Convert.ToUInt64(FunctionGroups.Sum(x => Convert.ToDecimal(x.ExecutionTimeInMS)))
+                                                  };
+
+                string siteName = GetNormalizedString(Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME"));
```

Comment:
Oh I didn't saw the comment on my commit. Maybe it was on Ashish's Change. Anyway, I'll do the change. Thanks for pointing this out.